p2p/pre2p/types/network.go

```diff
@@ -1,18 +1,40 @@
 package types
 
 import (
-	"net"
-
 	cryptoPocket "github.com/pokt-network/pocket/shared/crypto"
 )
 
+// CLEANUP(olshansky): See if we can deprecate one of these structures.
+type AddrBook []*NetworkPeer
+type AddrBookMap map[string]*NetworkPeer
+
+// TECHDEBT(olshansky): When we delete `stdnetwork` and only go with `raintree`, this interface
+// can be simplified greatly.
 type Network interface {
 	NetworkBroadcast(data []byte) error
 	NetworkSend(data []byte, address cryptoPocket.Address) error
-	GetAddrBook() []*NetworkPeer
+
+	// Address book helpers
+	GetAddrBook() AddrBook
+	AddPeerToAddrBook(peer *NetworkPeer) error    // TODO(team): Not used yet
+	RemovePeerToAddrBook(peer *NetworkPeer) error // TODO(team): Not used yet
+
+	// This function was added to specifically support the RainTree implementation.
+	// Handles the raw data received from the network and returns the data to be processed
+	// by the application layer.
+	HandleNetworkData(data []byte) ([]byte, error)
 }
 
 type NetworkPeer struct {
-	ConsensusAddr *net.TCPAddr
-	PublicKey     cryptoPocket.PublicKey
+	Dialer     TransportLayerConn
+	PublicKey  cryptoPocket.PublicKey
+	Address    cryptoPocket.Address
+	ServiceUrl string // This is only included because it's a more human-friendly differentiator between peers
+}
+
+type TransportLayerConn interface {
+	IsListener() bool
+	Read() ([]byte, error)
+	Write([]byte) error
+	Close() error
 }
```

Comment:
```suggestion
type Transport interface {
	IsListener() bool
	IsDialer() bool
	Accept() (net.Conn, error)
	Dial(...) (net.Conn, error)
	Close() error
}
```

Reply:
This is a suggestion, I think the name can definitely change, in terms of interface methods and behavior, I am not sure we can do it before #86 so we can keep it as is until then.

Reply:
1. I'm open to changing `TransportLayerConn` to `Transport`, but can you get a 2nd opinion from someone or a reference showing it's the way to go?

2. Do you think it's a good idea to start adding unused interface functions here before we have a need for them?

Reply:
Renamed `TransportLayerConn to Transport` and pushed.